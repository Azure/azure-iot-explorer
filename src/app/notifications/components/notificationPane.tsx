/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { Panel, PanelType } from 'office-ui-fabric-react/lib/components/Panel';
import { Link } from 'office-ui-fabric-react/lib/components/Link';
import { ActionButton } from 'office-ui-fabric-react/lib/components/Button';
import { ResourceKeys } from '../../../localization/resourceKeys';
import { ROUTE_PARTS, ROUTE_PARAMS } from '../../constants/routes';
import { useGlobalStateContext } from '../../shared/contexts/globalStateContext';
import '../../css/_headerPane.scss';

export const NotificationPane: React.FC = () => {
    const { t } = useTranslation();

    const [ showPanel, setShowPanel ] = React.useState<boolean>(false);
    const { globalState } = useGlobalStateContext();
    const { hasNew } = globalState.notificationsState;

    const togglePanelVisibility = () => {
        setShowPanel(!showPanel);
    };

    return (
        <>
            <ActionButton
                iconProps={{
                    className: hasNew && 'new-notifications',
                    iconName: 'Ringer'
                }}
                onClick={togglePanelVisibility}
                text={t(ResourceKeys.header.notifications.show)}
                ariaLabel={t(ResourceKeys.header.notifications.show)}
            />

            <Panel
                className="headerPane"
                role="dialog"
                isOpen={showPanel}
                isLightDismiss={true}
                onDismiss={togglePanelVisibility}
                type={PanelType.smallFixedFar}
                isFooterAtBottom={true}
                closeButtonAriaLabel={t(ResourceKeys.common.close)}
            >
                <header className="panel-header">
                    <h2>{t(ResourceKeys.header.notifications.panel.title)}</h2>
                </header>
                <section aria-label={t(ResourceKeys.header.notifications.panel.title)}>
                    <span>{t(ResourceKeys.header.notifications.panel.redirect)}</span>
                    <Link
                        className="home-link"
                        onClick={togglePanelVisibility}
                        href={`#/${ROUTE_PARTS.HOME}/${ROUTE_PARTS.NOTIFICATIONS}?${ROUTE_PARAMS.NAV_FROM}`}
                    >
                        {t(ResourceKeys.header.notifications.panel.redirectLink)}
                    </Link>
                </section>
            </Panel>
        </>
    );
};
