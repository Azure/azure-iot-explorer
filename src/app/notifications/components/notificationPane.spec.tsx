/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
 import 'jest';
 import * as React from 'react';
 import { shallow, mount } from 'enzyme';
 import { Panel, ActionButton } from '@fluentui/react';
 import { act } from 'react-dom/test-utils';
 import { NotificationPane } from './notificationPane';

 describe('notificationPane', () => {
    it('matches snapshot', () => {
        expect(shallow(<NotificationPane/>)).toMatchSnapshot();
    });

    it('toggles visibility', () => {
        jest.spyOn(React, 'useState').mockImplementationOnce(() => React.useState(true));
        const wrapper = mount(<NotificationPane/>);
        expect(wrapper.find(Panel).props().isOpen).toEqual(true);

        act(() => wrapper.find(ActionButton).first().props().onClick(null));
        wrapper.update()
        expect(wrapper.find(Panel).props().isOpen).toEqual(false);
    });
});
