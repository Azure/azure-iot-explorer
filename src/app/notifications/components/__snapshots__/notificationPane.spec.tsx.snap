// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notificationPane matches snapshot 1`] = `
<Fragment>
  <CustomizedActionButton
    ariaLabel="header.notifications.show"
    iconProps={
      Object {
        "className": false,
        "iconName": "Ringer",
      }
    }
    onClick={[Function]}
    text="header.notifications.show"
  />
  <StyledPanelBase
    className="headerPane"
    closeButtonAriaLabel="common.close"
    isFooterAtBottom={true}
    isLightDismiss={true}
    isOpen={false}
    onDismiss={[Function]}
    role="dialog"
    type={1}
  >
    <header
      className="panel-header"
    >
      <h2>
        header.notifications.panel.title
      </h2>
    </header>
    <section
      aria-label="header.notifications.panel.title"
    >
      <span>
        header.notifications.panel.redirect
      </span>
      <StyledLinkBase
        className="home-link"
        href="#/home/notifications?from"
        onClick={[Function]}
      >
        header.notifications.panel.redirectLink
      </StyledLinkBase>
    </section>
  </StyledPanelBase>
</Fragment>
`;
